metrics-service/src/main/java/org/ametiste/metrics/AggregatingMetricsService.java

```diff
@@ -72,10 +75,20 @@ public void gauge(String metricId, int gaugeValue) {
      */
     @Override
     public void createEvent(String metricId, int eventValue) {
-        router.getAggregatorsForMetric(metricId).forEach(metricAggregator ->
+        route(metricId, (metricAggregator) ->
                 metricAggregator.event(resolve(metricId), eventValue));
     }
 
+    private void route(String metricId, Consumer<MetricsAggregator> action) {
+        try {
+            router.aggregate(metricId, action);
+        }catch (Exception e) {
+            // do nothing, our service should not stop flow in common case
+            logger.warn("Router threw an exception", e);

```

Comment:
`WARN` level is so noisy for these purposes, `DEBUG` or `if (isDebugEnabled()) { warn(...) }` would be better.